Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/storage/memory/MemoryNode.java

```diff
@@ -0,0 +1,1096 @@
+     */
+    @Override
+    public MemoryResult find(Node graphPath) {
+        return (MemoryResult) GraphPath.apply(new MemoryResult(this), graphPath);
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @Override
+    public Optional<Long> first() {
+        Indices range = range();
+        return range == null ? Optional.empty() : Optional.of(range.first);
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @Override
+    public MemoryResult get(Collection<String> relations, Collection<ObjectType> conditions) {
+        MemoryResult result = new MemoryResult();
+        if (relations.isEmpty()) {
+            relations = new LinkedList<>(Arrays.asList(new String[] {GraphPath.ANY_PREDICATE.getIdentifier() }));
+        }
+        for (String relation : relations) {
+            Iterator<ObjectType> nodes = GraphPath.ANY_PREDICATE.getIdentifier().equals(relation) ? iterator()
+                    : edges.containsKey(relation) ? edges.get(relation).iterator()
+                            : Collections.<ObjectType>emptyIterator();
+            while (nodes.hasNext()) {
+                ObjectType node = nodes.next();
+                if (conditions.isEmpty()) {
+                    result.add(node);
+                } else {
+                    if (node instanceof AccessibleObject) {
+                        Collection<ObjectType> remainingConditions = new LinkedList<>(conditions);
+                        for (Iterator<ObjectType> i = remainingConditions.iterator(); i.hasNext();) {
+                            ObjectType condition = i.next();
+                            if (((AccessibleObject) node).matches(condition)) {
+                                i.remove();
+                            }
+                        }
+                        if (remainingConditions.isEmpty()) {
+                            result.add(node);
+                        }
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @Override
+    public Result get(IdentifiableNode relation, IdentifiableNode identifierRelation, String identifierValue) {
+        return get(relation != null ? relation.getIdentifier() : null, identifierRelation,
+            new MemoryLiteral(identifierValue, RDF.PLAIN_LITERAL));
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @Override
+    public MemoryResult get(long index) {
+        return new MemoryResult(get(RDF.toURL(index)));
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @Override
+    public MemoryResult get(String relation) {
+        return new MemoryResult(edges.get(relation));
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @Override
+    public MemoryResult get(String relation, IdentifiableNode identifierRelation, ObjectType identifierValue) {
+        return get(relation != null ? Arrays.asList(new String[] {relation }) : null,
```

Comment:
Return value `null` should be avoided. Is a `new MemoryResult()` not a better solution?